### PR TITLE
Feature/197 partial iso strings

### PR DESCRIPTION
**Pull request by chevalleyc**

### CODE REVIEW

File: utils/src/main/java/com/nedap/archie/datetime/DateTimeParsers.java
Comment (pieterbos) [inline utils/src/main/java/com/nedap/archie/datetime/DateTimeParsers.java:30]: The format in the try catch seems the same as this one. Could this try/catch be replaced with the following with the same result?

```
                return DateTimeFormatters.ISO_8601_DATE_TIME.parseBest(text, LocalDate::from, Year::from);
```

**Review by pieterbos — CHANGES_REQUESTED** [review]
Thanks for this pull request. I see just one small improvement.

**Comment by J3173** [comment]
The openEHR specification is a bit ambiguous at this point. 

The [Iso8601_date_time Class](https://specifications.openehr.org/releases/BASE/latest/foundation_types.html#_iso8601_date_time_class) suggests that all fields except year are optional. 

However the definition of `valid_iso8601_date_time` in the [Time_Definitions Class](https://specifications.openehr.org/releases/BASE/latest/foundation_types.html#_time_definitions_class) (referenced from the [DV_DATE_TIME Class](https://specifications.openehr.org/releases/RM/latest/data_types.html#_dv_date_time_class)) seems to imply that only seconds are optional and everything else is required.

File: utils/src/main/java/com/nedap/archie/datetime/DateTimeParsers.java
Comment (J3173) [inline utils/src/main/java/com/nedap/archie/datetime/DateTimeParsers.java:44]: Should this also be changed to support partial datetimes?

Comment (chevalleyc) [reply utils/src/main/java/com/nedap/archie/datetime/DateTimeParsers.java:44]: @J3173 indeed, but at the moment (and to make changes done in an acceptable timeframe) we have somehow decided to focus on non compact ISO8601 expression

File: utils/src/main/java/com/nedap/archie/datetime/DateTimeParsers.java
Comment (chevalleyc) [reply utils/src/main/java/com/nedap/archie/datetime/DateTimeParsers.java:30]: @pieterbos this has been a trial  and error process and this nested try {} catch {} finally ended to work.

Comment (pieterbos) [reply utils/src/main/java/com/nedap/archie/datetime/DateTimeParsers.java:30]: I'll create a branch and try later today, then show you the results - my suggestion should have the exact same behaviour and the tests should pass without modification, parseBest is equivalent to the try/catch here, but in less code.

Comment (pieterbos) [reply utils/src/main/java/com/nedap/archie/datetime/DateTimeParsers.java:30]: I created a pull request on your branch that does this at https://github.com/ehrbase/archie-fork/pull/1 :)

**Review by J3173 — APPROVED** [review]

**Comment by chevalleyc** [comment]
@pieterbos cannot merge the changes until you approve the Change Request :-(

**Review by pieterbos — APPROVED** [review]

**Comment by chevalleyc** [comment]
@pieterbos hehe... not the end of the long march with this PR. I don't have a write access to perform the merge, I believe you might be able to do it. Many thanks!

**Comment by pieterbos** [comment]
I approved it, but semaphore CI seems to have a problem here as well. No problem, https://openehr.semaphoreci.com/branches/1a282d25-940f-4e92-9de4-04e09b0d37f1 is the same code and runs fine. So I'll merge this.

I think we may need to add Jeltes suggestion for the compact format as well. I'll add an issue - should be simple to do, but it requires a bit more tests as well - don't have time now.

If you need a release of Archie for EHRBase, just send me a message or e-mail and we'll make sure to make a release.

**Pull request merged**
